test/unittest/tests/block/test_extent_list_search.cpp

```diff
@@ -0,0 +1,297 @@
+        std::vector<wt_off_t> test_list{0, 4096, 3 * 4096}; // 3, 4096, 12,288.
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+
+        /* Test. */
+        WT_EXT dummy;
+        for (const wt_off_t &test : test_list) {
+            INFO("Search: off " << test);
+            /* Set to an invalid value to determine whether __block_off_srch_pair changed them. */
+            WT_EXT *before = &dummy;
+            WT_EXT *after = &dummy;
+            /* Call. */
+            __ut_block_off_srch_pair(&extlist, test, &before, &after);
+            /* Verify: All should be not found. */
+            REQUIRE(before == nullptr);
+            REQUIRE(after == nullptr);
+        }
+    }
+
+    SECTION("search a non-empty list")
+    {
+        BREAK;
+        /* Extents to insert to create an extent list to search. */
+        std::vector<off_size> insert_list{
+          off_size(3 * 4096, 4096), // Second [12,288, 16,383].
+          off_size(4096, 4096),     // First [4,096, 8,191].
+          off_size(5 * 4096, 4096), // Third [20,480, 24,575].
+        };
+
+        /* Tests and expected values for __block_srch_pair. */
+        std::vector<search_before_after> expected_before_after{
+          search_before_after(0, nullptr, &insert_list[1]),    // Before first 0.
+          search_before_after(4096, nullptr, &insert_list[1]), // At first 4,096.
+          search_before_after(
+            2 * 4096, &insert_list[1], &insert_list[0]), // Between first and second 8,192.
+          search_before_after(3 * 4096, &insert_list[1], &insert_list[0]), // At second 12,288.
+          search_before_after(
+            4 * 4096, &insert_list[0], &insert_list[2]), // Between second and third 16,384.
+          search_before_after(5 * 4096, &insert_list[0], &insert_list[2]), // At third 20,480.
+          search_before_after(6 * 4096, &insert_list[2], nullptr),         // After third 24,576.
+        };
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+
+        /* Insert extents. */
+        for (const off_size &to_insert : insert_list) {
+
+            INFO("Insert: {off " << std::showbase << to_insert.off << ", size " << to_insert.size
+                                 << ", end " << to_insert.end() << "}");
+            REQUIRE(__ut_block_off_insert(session, &extlist, to_insert.off, to_insert.size) == 0);
+        }
+
+        extlist_print_off(extlist);
+
+        /* Test. */
+        WT_EXT dummy;
+        uint32_t idx = 0;
+        for (const search_before_after &expected : expected_before_after) {
+            WT_EXT *before = &dummy;
+            WT_EXT *after = &dummy;
+            /* Call. */
+            __ut_block_off_srch_pair(&extlist, expected.off, &before, &after);
+
+            std::ostringstream line_stream;
+            line_stream << "Verify: " << idx << ". off " << expected.off;
+            if (expected.before != nullptr)
+                line_stream << "; Expected: before: {off " << expected.before->off << ", size "
```

Comment:
You could simplify these lines by defining an << operator, e.g. something along the lines of:
```
std::ostream& operator<< (std::ostream& out, const off_size& s) {
    out << "{off " << s.off << ", size " << s.size << ", end " << s.end() << "}";
    return out;
}
```
and if possible also this (I don't remember right now if C++ likes an operator overload for pointers):
```
std::ostream& operator<< (std::ostream& out, const off_size* s) {
    if (s == nullptr)
        out << "nullptr";
    else
        out << *s;
    return out;
}
```
Then,  this would be just:
```
line_stream << "Expected: before: " << expected.before;
```
Similarly, there are many other lines in this file (including in the other tests) that this would simplify.

Reply:
I didn't do this since there are many more struct members than I am printing. I expected every place printing is done would have unique requirements of which members to print.

I have defined one print function.  If the members needed are different, people will still have to roll their own.

Done